### PR TITLE
feat(autoware_traffic_light_arbiter): add current time validation

### PR DESCRIPTION
**Pull request by MasatoSaeki**
## Description
This PR enables to validate external messages compared by current time.

## Related links

**Parent Issue:**
- [TIER IV INTERNAL LINK](https://tier4.atlassian.net/jira/software/c/projects/ISSJP/boards/671?assignee=6061477cafe465006afac8aa&selectedIssue=ISSJP-1798)
- https://github.com/autowarefoundation/autoware_launch/pull/1289

## How was this PR tested?
I have checked it on [Evaluator](https://evaluation.ci.tier4.jp/evaluation/reports/8ffde025-0720-59e9-9300-18d76379227b?project_id=prd_jt). 

## Notes for reviewers

None.

## Interface changes

None.

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added | `external_current_time_tolerance`   | double | 5.0           | The duration in seconds an external message is considered valid for merging in comparison with current time                                                                 |

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_traffic_light_arbiter/config/traffic_light_arbiter.param.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /**:
   ros__parameters:
+    external_current_time_tolerance: 5.0
```
Comment (YoshiRi) [inline perception/autoware_traffic_light_arbiter/config/traffic_light_arbiter.param.yaml:3]: How about `external_delay_tolerance` ?

**Review by YoshiRi — CHANGES_REQUESTED** [review]

File: perception/autoware_traffic_light_arbiter/config/traffic_light_arbiter.param.yaml (hunk 1 above)
Comment (MasatoSaeki) [reply perception/autoware_traffic_light_arbiter/config/traffic_light_arbiter.param.yaml]: It looks good to me. I fixed in https://github.com/autowarefoundation/autoware.universe/pull/9747/commits/227d53926f6802fe4532ca53b217e2205957cf85.

**Comment by knzo25** [comment]
I see changes that aim to address the sync issues between the V2X source and the perception stack (I did not consider that during the original implementation). To handle all the possible cases, could you make the comparisons using `std::abs` ?

**Comment by knzo25** [comment]
The tests seem to be failing. Maybe the stamps need to be generated during the test?

**Comment by MasatoSaeki** [comment]
@knzo25 
Thanks for your input.
> I see changes that aim to address the sync issues between the V2X source and the perception stack (I did not consider that during the original implementation). To handle all the possible cases, could you make the comparisons using std::abs ?

fixed in https://github.com/autowarefoundation/autoware.universe/pull/9747/commits/982643f04f3ae42bfaca6d5c102601ce71a6d08f

> The tests seem to be failing. Maybe the stamps need to be generated during the test?

fixed in https://github.com/autowarefoundation/autoware.universe/pull/9747/commits/e2916dddee7b6afcbfeafd86a6a31706108b2233

**Comment by knzo25** [comment]
@MasatoSaeki 

It seems there is a ci/cd error:
```
perception/autoware_traffic_light_arbiter/src/traffic_light_arbiter.cpp:241:  Add #include <algorithm> for max  [build/include_what_you_use] [4]
Done processing perception/autoware_traffic_light_arbiter/src/traffic_light_arbiter.cpp
```

**Comment by MasatoSaeki** [comment]
@knzo25 
I fixed in https://github.com/autowarefoundation/autoware.universe/pull/9747/commits/0e85287308753ca06275ffec78bff781ba732271

**Comment by MasatoSaeki** [comment]
@YoshiRi 
Could you check this PR regarding change requested?

**Review by YoshiRi — APPROVED** [review]
LGTM

**Pull request merged**